stl/inc/vector

```diff
@@ -1275,17 +1297,27 @@ private:
         _Change_array(_Newvec, _Size, _Newcapacity);
     }
 
-    void _Clear_and_reserve_geometric(const size_type _Newsize) {
+    void _Check_all_orphaned_locked() {
+        _Lockit _Lock(_LOCK_DEBUG);
+        auto& _My_data = _Mypair._Myval2;
+        _STL_INTERNAL_CHECK(!_My_data._Myproxy->_Myfirstiter);

```

Comment:
The first test failure in the logs is here. This is ill-formed when `_ITERATOR_DEBUG_LEVEL == 0`, in which case `_My_data` has no member named `_Myproxy`.

Reply:
I think the problem was that I pulled out the Check into its own function without defining `auto& _My_data = _Mypair._Myval2;

Note that `_Check_all_orphaned_locked()` is only ever called inside of `#if _ITERATOR_DEBUG_LEVEL != 0 && defined(_ENABLE_STL_INTERNAL_CHECK)`

Reply:
(sorry for force pushing over the initial version again )

bad habbits ...

Reply:
> Note that `_Check_all_orphaned_locked()` is only ever called inside of `#if _ITERATOR_DEBUG_LEVEL != 0 && defined(_ENABLE_STL_INTERNAL_CHECK)`

That may be the intent, but the error in the test log disagrees. I suggest a full local run of `tests\std\tests\Dev11_0483851_vector_debug_allocator_use`.

Reply:
Yeah, I checked it forvector and this was the one missing case.

BTW there is a happy mix of `_ITERATOR_DEBUG_LEVEL == 2` and `_ITERATOR_DEBUG_LEVEL != 0` Is one of those preferred?

Reply:
> BTW there is a happy mix of `_ITERATOR_DEBUG_LEVEL == 2` and `_ITERATOR_DEBUG_LEVEL != 0` Is one of those preferred?

Go for local consistency. I've been preferring `!= 0` in new code (as I'm sure you're aware) since `_IDL1` is on the most minimal of life support waiting for us to pull the plug.

Reply:
Done